frontends/uhdm/UhdmAst.cc

```diff
@@ -555,7 +568,33 @@ void UhdmAst::process_module() {
 							  obj_h,
 							  [&](AST::AstNode* node) {
 								  if (node) {
+									if (node->type == AST::AST_INITIAL) {
+										auto module_child = current_node->children.begin();
+										for (;module_child != current_node->children.end(); module_child++) {
+											if ((*module_child)->type == AST::AST_ALWAYS) {
+												break;
+											}
+										}
+										if (module_child != current_node->children.end())
+										// Place AST_INITIAL before AST_ALWAYS
+											current_node->children.insert(module_child, 1, node);
+										else
+											add_or_replace_child(current_node, node);
```

Comment:
Isn't this already done in ``visit_default_expr``? If we already added node, we should return ``nullptr`` and here we will skip adding this node.

Reply:
We need those modifications in `process_module` because this is the place where we can control the node ordering in AST. When this peace of code is deleted, there would be `AST_ALWAYS` with whole *if* statement at the begining of  `AST_MODULE` and after that there would be `AST_ASSIGN` with `a` and `AST_INITIAL` with `b`. Such situation is messing up the whole AST which results in bad behavior of if statements in Yosys.